src/lib.rs

```diff
@@ -2402,7 +2404,8 @@ impl Build {
             let dlink = out_dir.join(lib_name.to_owned() + "_dlink.o");
             let mut nvcc = self.get_compiler().to_command();
             nvcc.arg("--device-link").arg("-o").arg(&dlink).arg(dst);
-            run(&mut nvcc, "nvcc", &self.cargo_output)?;
+            let program = env::var("CUDACXX").unwrap_or_else(|_| String::from("nvcc"));
```

Comment:
I think you might need to modify `cc::Build::get_base_compiler`.

This line you modified is for linking with cuda-specific part only.

P.S. I'm not so familiar with the cuda compilation part, so feel free to point out any mistake.

Reply:
I suspected this was too easy. I'll take a look.